channel/player.go

```diff
@@ -1387,6 +1401,39 @@ func (d *Player) setInventorySlotSizes(equip, use, setup, etc, cash byte) {
 	d.MarkDirty(DirtyInvSlotSizes, 2*time.Second)
 }
 
+func (d *Player) IncreaseSlotSize(invID, amount byte) error {
+	switch invID {
+	case constant.InventoryEquip:
+		if d.equipSlotSize+amount > 255 {
+			return fmt.Errorf("cannot increase equip slot size beyond %d", 255)
+		}
+		d.equipSlotSize += amount
+	case constant.InventoryUse:
+		if d.useSlotSize+amount > 255 {
+			return fmt.Errorf("cannot increase use slot size beyond %d", 255)
+		}
+		d.useSlotSize += amount
+	case constant.InventorySetup:
+		if d.setupSlotSize+amount > 255 {
+			return fmt.Errorf("cannot increase setup slot size beyond %d", 255)
+		}
+		d.setupSlotSize += amount
+	case constant.InventoryEtc:
+		if d.etcSlotSize+amount > 255 {
+			return fmt.Errorf("cannot increase etc slot size beyond %d", 255)
+		}
+		d.etcSlotSize += amount
+	case constant.InventoryCash:
+		if d.cashSlotSize+amount > 255 {
+			return fmt.Errorf("cannot increase cash slot size beyond %d", 255)
+		}
+		d.cashSlotSize += amount
+	}
+
+	d.MarkDirty(DirtyInvSlotSizes, 2*time.Second)
```

Comment:
Is there a chance a player can return to channel server before this update gets triggered?